test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,661 @@ describe("BaseAxis", () => {
+
+          annotationLabelTexts.each(function (d, i) {
+            let annotationLabel = d3.select(this);
+            let surroundingRect = d3.select(annotationRects[0][i]);
+            TestMethods.assertBBoxInclusion(surroundingRect, annotationLabel);
+          });
+          svg.remove();
+        });
+
+        it("renders annotations at the axis baseline extending upwards with a top orientation axis", () => {
+          let axis = new Plottable.Axis(scale, "top");
+          axis.annotatedTicks(annotatedTicks);
+          axis.annotationsEnabled(true);
+
+          let svg = TestMethods.generateSVG(300, 300);
+          axis.renderTo(svg);
+
+          let annotationRects = axis.content().selectAll(".annotation-rect");
+
+          let annotationCircles = axis.content().selectAll(".annotation-circle");
+
+          assert.strictEqual(annotationCircles.size(), annotationRects.size(), "same number of annotation circles as rects");
+
+          annotationCircles.each(function (d, i) {
+            let annotationCircle = d3.select(this);
+            let correspondingRect = d3.select(annotationRects[0][i]);
+            assert.closeTo(TestMethods.numAttr(annotationCircle, "cx"),
+              TestMethods.numAttr(correspondingRect, "x"), window.Pixel_CloseTo_Requirement, `circle ${i} at start of rect`);
+            assert.closeTo(TestMethods.numAttr(annotationCircle, "cy"), axis.height(),
+              window.Pixel_CloseTo_Requirement, `circle ${i} at the bottom`);
+          });
+
+          let annotationLines = axis.content().selectAll(".annotation-line");
+
+          assert.strictEqual(annotationLines.size(), annotationRects.size(), "same number of annotation lines as rects");
+
+          annotationLines.each(function (d, i) {
+            let annotationLine = d3.select(this);
+            let correspondingRect = d3.select(annotationRects[0][i]);
+            assert.closeTo(TestMethods.numAttr(annotationLine, "x1"),
+              TestMethods.numAttr(correspondingRect, "x"), window.Pixel_CloseTo_Requirement, `line ${i} at tick scaled x position`);
+            assert.closeTo(TestMethods.numAttr(annotationLine, "x2"),
+              TestMethods.numAttr(correspondingRect, "x"), window.Pixel_CloseTo_Requirement, `line ${i} at tick scaled x position`);
+            assert.closeTo(TestMethods.numAttr(annotationLine, "y1"), axis.height(),
+              window.Pixel_CloseTo_Requirement, `line ${i} starts at the bottom`);
+            assert.closeTo(TestMethods.numAttr(annotationLine, "y2"),
+              TestMethods.numAttr(correspondingRect, "y") + TestMethods.numAttr(correspondingRect, "height"),
+              window.Pixel_CloseTo_Requirement, `line ${i} goes to the rect`);
+          });
+
+          let annotationLabelTexts = axis.content().selectAll(".annotation-label text");
+
+          assert.strictEqual(annotationLabelTexts.size(), annotationRects.size(), "same number of annotation labels as rects");
```

Comment:
Probably better to assert that each selection has the same `size()` as the number of ticks.